src/run_management.nim

```diff
@@ -198,53 +198,49 @@ proc newChalk*(name:         string            = "",
   if extract != nil and len(extract) > 1:
     result.marked = true
 
-template setIfNotEmpty*(dict: ChalkDict, key: string, val: string) =
-  if val != "":
-    dict[key] = pack(val)
-
-template setIfNotEmpty*[T](dict: ChalkDict, key: string, val: seq[T]) =
-  if len(val) > 0:
-    dict[key] = pack[seq[T]](val)
-
-template setFromEnvVar*(dict: ChalkDict, key: string, default: string = "") =
-  dict.setIfNotEmpty(key, os.getEnv(key, default))
+template setIfNotEmptyBox*(o: ChalkDict, k: string, v: Box) =
+  case v.kind
+  of MkSeq, MkTable, MkStr:
+    if len(v) > 0:
+      o[k] = v
+  else:
+    o[k] = v
+
+template setIfNotEmpty*[T](o: ChalkDict, k: string, v: T) =
+  when T is Box:
+    setIfNotEmptyBox(o, k, v)
+  elif T is JsonNode:
+    if v != nil:
+      setIfNotEmptyBox(o, k, v.nimJsonToBox())
+  elif T is Option:
+    if v.isSome():
+      setIfNotEmptyBox(o, k, pack(v.get()))
+  else:
+    setIfNotEmptyBox(o, k, pack(v))
```

Comment:
is this better as compound if statements to eliminate the nested conditionals?

Reply:
the outer is compile-time conditional. note the `when`vs `if`

https://nim-docs.readthedocs.io/en/latest/manual/statements_expressions/#when-statement

note that `when` operates on a generic type `T` here which is not available at run-time as its a compile-time construct

for example another common use-case for `when` is for debug-related things:

https://github.com/crashappsec/chalk/blob/81c0d04646a1ed9aee623f13edf469e661d3b2a9/src/chalk_common.nim#L507-L517

when doing release builds, that code-block is completely omitted from the binary